modules/dhcp/providers/server/native_ms.rb

```diff
@@ -196,6 +230,61 @@ def execute cmd, msg=nil, error_only=false
       response
     end
 
+    # Invokes netsh.exe with a script file
+    def execute_with_script scriptfile, msg=nil, error_only=false
+      # this should be configurable elsewhere - setting to 30 to allow for latency across slow links
+      tsecs = 30
+      response = nil
+      interpreter = Proxy::SETTINGS.x86_64 ? 'c:\windows\sysnative\cmd.exe' : 'c:\windows\system32\cmd.exe'
+      command  = interpreter + ' /c c:\Windows\System32\netsh.exe' + " -r #{name} -f #{scriptfile}"
+
+      std_in = std_out = std_err = nil
+      begin
+        timeout(tsecs) do
+          logger.debug "executing: #{command}"
+          std_in, std_out, std_err  = Open3.popen3(command)
+          response  = std_out.readlines
+          response += std_err.readlines
+        end
+      rescue TimeoutError
+        raise puts("Netsh did not respond within #{tsecs} seconds")
+      ensure
+        std_in.close  unless std_in.nil?
+        std_out.close unless std_in.nil?
+        std_err.close unless std_in.nil?
+      end
+      report msg, response, error_only
+      response
+    end
+
+    # convert scriptfile output into a hash for fast queries
+    def parse_script_output_to_hash content
+      processed = {}
+      queryResults = ""
+      count = 0
+      found = false
+      anIP = nil
+      content.each do |line|
```

Comment:
I guess this is the line Oliver's referring to here: http://projects.theforeman.org/issues/7766#change-44781